Microphones are sending mac address

nebojsarajic opened the pull request:
When microphone is turned on, it will send mac address in message. If it's not in collection, it will be added, otherwise removed. Files are going to be saved with mac address in name, so we can determine which microphone recorded specific audio file.
Adding support for windows OS. Fixing hard-coded paths.

Commit:
Sending mac address from microphones when they start

Commit:
Fixing paths and enabling server and microphone on Windows OS

chedadsp commented on audio_recorder/src/main.py:
```diff
@@ -14,10 +14,10 @@
+    output_file_path = os.path.join(os.path.dirname(os.path.abspath(__file__)), "..", "output", "test.wav")
 
     udp_sender = UDPSender()
-    udp_sender.starting_server_message()
```
I don't like that we do not have start/stop messages any more. In a case where audio_recodred fail for some reason we can have invalid status by just toggling presence. This error condition is not checked elsewhere so we can have follow up problems. Please fix.

chedadsp commented on audio_recorder/src/main.py:
```diff
@@ -14,10 +14,10 @@
     recorder = Recorder()
     audio_properties = Audio_properties()
-    output_file_path = "../output/test.wav"
+    output_file_path = os.path.join(os.path.dirname(os.path.abspath(__file__)), "..", "output", "test.wav")
```
I like that we os sensitive routines to calculate path. But, logic is still the same. We are assuming that output folder is created and that it is on one level up from this routine. this can be problematic for real deployment. 

I think it would be little bit better if we store file in "current folder".
Even better, in audio_recorder app we don't have to use file system at all. Recorded sound can be stored in stream in memory and served to main server when requested.

chedadsp commented on main_server/src/main.py:
```diff
@@ -32,18 +33,18 @@ def index_page():
-            response = requests.get('http://{}:63000/record'.format(microphone))
-            check_for_response(response, microphone)
+        for microphone_ip, _ in udp_receiver.get_microphones().items():
+            response = requests.get('http://{}:63000/record'.format(microphone_ip))
```
Same literal used in multiple places should be a constant.

chedadsp reviewed (changes requested):
Please see my comment and make changes.

Commit:
Fixing comments from pull request

Microphones aren't using file system anymore. It's saving audio in memory
and sending as byte array in response.
Port for microphones is now constant.
Returning start and stop message for microphones.
